internal/service/s3/object.go

```diff
@@ -149,6 +149,10 @@ func resourceObject() *schema.Resource {
 				Computed:      true,
 				ConflictsWith: []string{names.AttrKMSKeyID},
 			},
+			"if_none_match": {
```

Comment:
Quick drive-by comment, setting the value to `*` will break the PutObject operation in a subsequent call to resource.Create, according to the documentation:

> If there is an existing object, then the write operation fails.